### ISSUE TITLE
Krakend plugin not getting registered when i run krakend container

### ISSUE THREAD

**Issue by amokkara** [krakend/krakend-ce#865]
**Environment info:**
Running krakend docker container on ubuntu v20.4
docker base image - golang:1.22.0-alpine3.19

**Describe what are you trying to do**:
Hi, currently I'm trying to run a krakend container along with a plugin that i created.
I'm able to successfully build the docker image but when i bring up the container, i notice the plugin is not getting registered.
I think it has something to do with base image used in Dockerfile and golang version used to build krakend. Could some one please take a look at tell me what's wrong with this configuration.

Dockerfile:

` #syntax = docker/dockerfile:1.2
ARG BASE_IMAGE="golang:1.22.0-alpine3.19"

FROM ${BASE_IMAGE} as builder
RUN apk add --no-cache --virtual .build-deps \
  gcc \
  libc-dev \
  make \
  && rm -rf /var/cache/apk/* \
RUN apk update && apk add --no-cache ca-certificates
WORKDIR /src

#
# Cache external dependencies for subsequent builds
#
COPY go.mod .
COPY go.sum .

ARG goproxy
ENV GOPROXY=$goproxy

RUN go mod download
 
COPY main.go .
COPY Makefile .

ARG next_image_tag
RUN --mount=type=cache,target=/root/.cache/go-build make NEXT_IMAGE_TAG=$next_image_tag docker-build

#
# Assemble the application
#
FROM registry.access.redhat.com/ubi8-minimal:8.7-1049
RUN microdnf install yum \
  && rpm -Uvh https://repo.krakend.io/rpm/krakend-repo-0.2-0.x86_64.rpm \
  && yum install -y krakend-2.6.0 \
  && yum clean all \
  && rm -f /etc/dnf/protected.d/yum.conf \
  && microdnf remove yum \
  && microdnf clean all

# Verify the plugin. If this validation fails, it will fail the build.
COPY --from=builder /src/go.sum /plugins/
RUN krakend check-plugin --go 1.20.14 --libc "GLIBC-2.31_(debian-11)" -s /plugins/go.sum
RUN rm -rf /plugins/go.sum

COPY --from=builder /src/bin/* /plugins/
COPY krakend.json /etc/krakend/krakend.json
COPY partials /etc/partials

ENTRYPOINT [ "krakend" ]
CMD [ "run", "-c", "/etc/krakend/krakend.json" ]
`

**Comment by taik0** [comment]
Hi @amokkara 

You are building a plugin for Docker+Alpine and using it on a KrakenD binary built for Debian/Rhel.
You have 2 options:
1- Build the plugin for Debian/Rhel
2- Use the KrakenD container (Alpine)

In any case, you should build the plugin using the KrakenD builder containers that already have the required versions and software:

https://hub.docker.com/r/krakend/builder

Check this Makefile with instructions for each possible build:
```
# KrakenD Version
VERSION := latest
PLUGINAME := plugin

# Build KrakenD plugin using the KrakenD docker builder (for running inside docker)
amd64:
	docker run --rm -it -e CGO_ENABLED=1 -e GOARCH=amd64 -e ARCH=amd64 -v "${PWD}:/app" -w /app krakend/builder:${VERSION} sh -c "make -e plugin"

arm64:
	docker run --rm -it -e CGO_ENABLED=1 -e CC=aarch64-linux-musl-gcc -e GOHOSTARCH=amd64 -e GOARCH=arm64 -e ARCH=arm64 -e EXTRA_LDFLAGS='-extld=aarch64-linux-musl-gcc' -v "${PWD}:/app" -w /app krakend/builder:${VERSION} sh -c "make -e plugin"

# Build KrakenD plugin using the KrakenD docker builder for non-docker plugins
linux_amd64:
	docker run --rm -it -e CGO_ENABLED=1 -e GOARCH=amd64 -e ARCH=amd64 -v "${PWD}:/app" -w /app krakend/builder:${VERSION}-linux-generic sh -c "make -e plugin"

linux_arm64:
	docker run --rm -it -e CGO_ENABLED=1 -e CC=aarch64-linux-gnu-gcc -e ARCH=arm64 -e GOARCH=arm64 -e GOHOSTARCH=amd64 -e EXTRA_LDFLAGS='-extld=aarch64-linux-gnu-gcc' -v "${PWD}:/app" -w /app krakend/builder:${VERSION}-linux-generic sh -c "make -e plugin"

plugin:
	@go build -buildmode=plugin -ldflags="${EXTRA_LDFLAGS}" -o ${PLUGINAME}.so .
```

**Comment by amokkara** [comment]
Hi @taik0 ,
Thanks for the clarification.  I tried with latest bullseye golang image 

this is what i get as error when try to build an image 


` => ERROR [stage-1 5/9] RUN krakend check-plugin --go 1.22 --libc "GLIBC-2.31_(debian-11)" -s /plugins/go.sum                  0.3s
------
 > [stage-1 5/9] RUN krakend check-plugin --go 1.22 --libc "GLIBC-2.31_(debian-11)" -s /plugins/go.sum:
0.264 1 incompatibility(ies) found...
0.264 go
0.264   have: 1.22
0.264   want: 1.20.14
------
Dockerfile:53
--------------------
  51 |     # Verify the plugin. If this validation fails, it will fail the build.
  52 |     COPY --from=builder /src/go.sum /plugins/
  53 | >>> RUN krakend check-plugin --go 1.22 --libc "GLIBC-2.31_(debian-11)" -s /plugins/go.sum
  54 |     RUN rm -rf /plugins/go.sum
  55 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c krakend check-plugin --go 1.22 --libc \"GLIBC-2.31_(debian-11)\" -s /plugins/go.sum" did not complete successfully: exit code: 1
make: *** [Makefile:115: docker] Error 1`


if i change the "krakend check-plugin --go 1.22" to "krakend check-plugin --go 1.20.14" as suggested, my krakend container is not coming up. Its unable to register the plugin. 

i see this in docker logs 

`
Parsing configuration file: /etc/krakend/krakend.json
[KRAKEND] 2024/04/09 - 16:02:20.539 ▶ INFO Starting KrakenD v2.6.0
[KRAKEND] 2024/04/09 - 16:02:20.541 ▶ INFO Working directory is /
fatal error: runtime: no plugin module data

goroutine 1 [running]:
runtime.throw({0x33a0457?, 0x115db3b?})
        /usr/local/go/src/runtime/panic.go:1047 +0x5d fp=0xc001026bf8 sp=0xc001026bc8 pc=0x112d03d
plugin.lastmoduleinit()
        /usr/local/go/src/runtime/plugin.go:20 +0x8b0 fp=0xc001026ce0 sp=0xc001026bf8 pc=0x115d970
plugin.open({0xc00138c180, 0x1e})
        /usr/local/go/src/plugin/plugin_dlopen.go:77 +0x3ef fp=0xc001026f50 sp=0xc001026ce0 pc=0x187336f
plugin.Open(...)
        /usr/local/go/src/plugin/plugin.go:80
github.com/luraproject/lura/v2/transport/http/client/plugin.defaultPluginOpener({0xc00138c180?, 0xc001384b20?})
        /go/pkg/mod/github.com/luraproject/lura/v2@v2.6.1/transport/http/client/plugin/plugin.go:116 +0x1e fp=0xc001026f70 sp=0xc001026f50 pc=0x26ddefe
github.com/luraproject/lura/v2/transport/http/client/plugin.open({0xc00138c180?, 0x1f?}, 0x1?, {0x3bbb590?, 0xc00130be90})
        /go/pkg/mod/github.com/luraproject/lura/v2@v2.6.1/transport/http/client/plugin/plugin.go:83 +0x9d fp=0xc001027008 sp=0xc001026f70 pc=0x26ddbdd
github.com/luraproject/lura/v2/transport/http/client/plugin.load({0xc001384b20?, 0x1, 0xc000f36087?}, 0x3?, {0x3bbb590, 0xc00130be90})
        /go/pkg/mod/github.com/luraproject/lura/v2@v2.6.1/transport/http/client/plugin/plugin.go:58 +0xeb fp=0xc001027100 sp=0xc001027008 pc=0x26dd90b
github.com/luraproject/lura/v2/transport/http/client/plugin.LoadWithLogger({0xc000f36090?, 0xc001384b00?}, {0xc000f36087?, 0xc00133efc0?}, 0x4?, {0x3bbb590, 0xc00130be90})
        /go/pkg/mod/github.com/luraproject/lura/v2@v2.6.1/transport/http/client/plugin/plugin.go:51 +0x5d fp=0xc001027148 sp=0xc001027100 pc=0x26dd7bd
github.com/krakendio/krakend-ce/v2.LoadPluginsWithContext({0x3bb1558, 0xc000ea71d0}, {0xc000f36090, 0x9}, {0xc000f36087, 0x3}, {0x3bbb590, 0xc00130be90})
        /github/workspace/plugin.go:28 +0xc5 fp=0xc0010271a0 sp=0xc001027148 pc=0x2be1c85
github.com/krakendio/krakend-ce/v2.pluginLoader.LoadWithContext(...)
        /github/workspace/plugin.go:78
github.com/krakendio/krakend-ce/v2.(*pluginLoader).LoadWithContext(0x4f43cb8?, {0x3bb1558?, 0xc000ea71d0?}, {0xc000f36090?, 0xc2?}, {0xc000f36087?, 0x0?}, {0x3bbb590?, 0xc00130be90?})
        <autogenerated>:1 +0x6f fp=0xc0010271f0 sp=0xc0010271a0 pc=0x2be55af
github.com/krakendio/krakend-ce/v2.(*ExecutorBuilder).NewCmdExecutor.func1({{0xc000f36078, 0x4}, {0xc000ec6700, 0xc2, 0xc2}, {0x0, 0x0, 0x0}, 0x6fc23ac00, 0x45d964b800, ...})
`

**Comment by taik0** [comment]
TLDR: Use the [KrakenD Builder container](https://hub.docker.com/r/krakend/builder) that matches your KrakenD version, it already has all the required GOLANG version and OS libs.
